lib/Runtime/Base/FunctionBody.h

```diff
@@ -1159,10 +1161,48 @@ namespace Js
     //
     class FunctionProxy : public FunctionInfo
     {
+    public:
+        typedef RecyclerWeakReference<DynamicType> FunctionTypeWeakRef;
+        typedef JsUtil::List<FunctionTypeWeakRef*, Recycler, false, WeakRefFreeListedRemovePolicy> FunctionTypeWeakRefList;
+
     protected:
         FunctionProxy(JavascriptMethod entryPoint, Attributes attributes, int nestedCount, int derivedSize,
             LocalFunctionId functionId, ScriptContext* scriptContext, Utf8SourceInfo* utf8SourceInfo, uint functionNumber);
         DEFINE_VTABLE_CTOR_NO_REGISTER(FunctionProxy, FunctionInfo);
+
+        enum class AuxPointerType : uint8 {
+            deferredStubs = 0,
+            cachedSourceString = 1,
+            asmJsFunctionInfo = 2,
+            asmJsModuleInfo = 3,
+            pStatementMaps = 4,
+            stackNestedFuncParent = 5,
+            simpleJitEntryPointInfo = 6,
+            functionObjectTypeList = 7,           // Script function types not including the deferred prototype type
+            codeGenGetSetRuntimeData = 8,
+            propertyIdOnRegSlotsContainer = 9,    // This is used for showing locals for the current frame.
+            loopHeaderArray = 10,
+            codeGenRuntimeData = 11,
+            polymorphicInlineCachesHead = 12,     // DList of all polymorphic inline caches that aren't finalized yet
+            propertyIdsForScopeSlotArray = 13,    // For SourceInfo
+            polymorphicCallSiteInfoHead  = 14,
+            auxBlock = 15,                        // Optional auxiliary information
+            auxContextBlock = 16,                 // Optional auxiliary context specific information
+            referencedPropertyIdMap = 17,
+            literalRegexes = 18,
+            objLiteralTypes = 19,
+            scopeInfo = 20,
+
+            max,
+            invalid = 0xff
```

Comment:
nit: these should all be in pascal case as well.

Otherwise this change looks great to me!